internal/service/autoscaling/policy.go

```diff
@@ -793,3 +1163,81 @@ func flattenPredefinedMetricPairSpecification(predefinedMetricPairSpecification
 	predefinedMetricPairSpecificationFlat["resource_label"] = aws.StringValue(predefinedMetricPairSpecification.ResourceLabel)
 	return []map[string]interface{}{predefinedMetricPairSpecificationFlat}
 }
+
+func flattenCustomizedScalingMetricSpecification(customizedScalingMetricSpecification *autoscaling.PredictiveScalingCustomizedScalingMetric) []map[string]interface{} {
+	customizedScalingMetricSpecificationFlat := map[string]interface{}{}
+	if customizedScalingMetricSpecification == nil {
+		return []map[string]interface{}{customizedScalingMetricSpecificationFlat}
+	}
+	customizedScalingMetricSpecificationFlat["metric_data_queries"] = flattenMetricDataQueries(customizedScalingMetricSpecification.MetricDataQueries)
+	return []map[string]interface{}{customizedScalingMetricSpecificationFlat}
+}
+
+func flattenCustomizedLoadMetricSpecification(customizedLoadMetricSpecification *autoscaling.PredictiveScalingCustomizedLoadMetric) []map[string]interface{} {
+	customizedLoadMetricSpecificationFlat := map[string]interface{}{}
+	if customizedLoadMetricSpecification == nil {
+		return []map[string]interface{}{customizedLoadMetricSpecificationFlat}
+	}
+	customizedLoadMetricSpecificationFlat["metric_data_queries"] = flattenMetricDataQueries(customizedLoadMetricSpecification.MetricDataQueries)
+	return []map[string]interface{}{customizedLoadMetricSpecificationFlat}
+}
+
+func flattenCustomizedCapacityMetricSpecification(customizedCapacityMetricSpecification *autoscaling.PredictiveScalingCustomizedCapacityMetric) []map[string]interface{} {
+	customizedCapacityMetricSpecificationFlat := map[string]interface{}{}
+	if customizedCapacityMetricSpecification == nil {
+		return []map[string]interface{}{customizedCapacityMetricSpecificationFlat}
+	}
+	customizedCapacityMetricSpecificationFlat["metric_data_queries"] = flattenMetricDataQueries(customizedCapacityMetricSpecification.MetricDataQueries)
+
+	return []map[string]interface{}{customizedCapacityMetricSpecificationFlat}
+}
+
+func flattenMetricDataQueries(metricDataQueries []*autoscaling.MetricDataQuery) []interface{} {
+	metricDataQueriesFlat := map[string]interface{}{}
+	if metricDataQueriesFlat == nil {
+		return []interface{}{}
+	}
+
+	metricDataQueriesSpec := make([]interface{}, len(metricDataQueries))
+	for i := range metricDataQueriesSpec {
+		metricDataQuery := map[string]interface{}{}
+		rawMetricDataQuery := metricDataQueries[i]
+		metricDataQuery["id"] = aws.StringValue(rawMetricDataQuery.Id)
+		if rawMetricDataQuery.Expression != nil {
```

Comment:
Small style nit: for the optional arguments like `Expression`, `MetricStat`, `Label` and `ReturnData`. These can be sorted as well to make it easier to verify that all the arguments are in-place